testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jsonb/basic/JsonBindingDebugLoggingTest.java

```diff
@@ -0,0 +1,325 @@
+    *
+    * In this scenario the exception originates in JsonBindingProvider.readFrom.
+    * The exception is passed up the call stack to ClientResponse.readEntity which
+    * catches it as a RuntimeException.  The original exception is returned to the
+    * caller on the client side and this test logs the exception in the test's
+    * server.log for reference.
+    */
+   @Test
+   public void exceptionDuringClientReceive() throws Exception {
+      // perform request
+      WebTarget base = client.target(generateURL("/get/ok"));
+      Response response = base.request().get();
+
+      // count log messages before request
+      LogCounter errorStringLog = new LogCounter("ERROR",
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter resteasyExceptionLog = new LogCounter(
+              ".*ERROR .* RESTEASY004672.*", true,
+              ContainerConstants.DEFAULT_CONTAINER_QUALIFIER, true);
+
+
+      LogCounter yassonExceptionLog = new LogCounter(
+              "Caused by: javax.json.bind.JsonbException",
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter yassonStacktraceLog = new LogCounter("at org.eclipse.yasson",
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter applicationExcpetionLog = new LogCounter(
+              "Caused by: java.lang.RuntimeException: "
+            + JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName(),
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+
+      // use JsonBindingProvider to get exception
+      try {
+         response.readEntity(JsonBindingDebugLoggingItemCorruptedSet.class);
+         Assert.fail("Client doesn't throw Exception during reading of corrupted data");
+      } catch (ProcessingException e) {
+         StringWriter errors = new StringWriter();
+         e.printStackTrace(new PrintWriter(errors));
+         String stackTrace = errors.toString();
+
+         Assert.assertThat("Stracktrace doesn't contain javax.json.bind.JsonbException", stackTrace,
+               containsString("javax.json.bind.JsonbException"));
+         Assert.assertThat("Stracktrace doesn't contain yasson part", stackTrace,
+               containsString("org.eclipse.yasson.internal"));
+         Assert.assertThat("Stracktrace doesn't contain application exception", stackTrace,
+               containsString("Caused by: java.lang.RuntimeException: "
+                       + JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName()));
+      }
+
+      // assert log messages after request
+      Assert.assertThat("Application Exception should be logged",
+              applicationExcpetionLog.count(), is(1));
+      Assert.assertThat("Yasson exception should be logged",
+              yassonExceptionLog.count(), is(1));
+      Assert.assertThat("There should be no Yasson exception stacktrace logged",
+              yassonStacktraceLog.count(), greaterThan(0));
+   }

```

Comment:
What about to add assert that there is no Error log messages (as I suggested in my original commit)?